Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,7 +1,67 @@
-function setAlarm() {}
+function setAlarm() {
+  //Converting the time of now value to seconds from milliseconds
+  //Getting the time value from the input field, converting it into number, this will be in seconds
+  const alarmTime = Number(document.getElementById("alarmSet").value);
+  
+   document.getElementById('timeRemaining').innerText="Time Remaining: "+timeFormatter(alarmTime);
+   let currentTime=0;
 
-// DO NOT EDIT BELOW HERE
+  const checkAlarm=setInterval(() => {
+    
+    //Calculating current time inside the interval
+     currentTime++;
+
+    //Calculating the difference between the alarm time and the current time
+    const diff = alarmTime - currentTime;
+
+    //Updating the time remaining text in the HTML element
+    document.getElementById('timeRemaining').innerText="Time Remaining: "+timeFormatter(diff);
+
+    
+
+    //If the difference is less than or equal to zero, stop the interval and play the alarm sound
+    if(diff <= 0){
+      clearInterval(checkAlarm);
+      playAlarm();
+      
+      
+    }
+  }, 1000);
+}
+
+//formatting remaining seconds in the format of MM:SS
+function timeFormatter(seconds) {
```

Comment:
We normally name functions like verbs because they _do_ things - so `formatTime` rather than `timeFormatter`. (Same comment applies to `timePadder`)